### ISSUE TITLE
Decimals

### ISSUE THREAD

**Issue by agenceldp** [WPprodigy/WooCommerce-Product-Fees#57]
Hello,

I'm a french freelance and first of all thank you very much for your plug-in !

I installed the plug-in on a wordpress site and I have a problem. 
The site in pogress sells goods in consignment. Also, the plugin is used to enter a fee-consignment per product added to the cart.

For example a milk bottle, it is 0.50€ euro deposit.

The total does not take into account the amount of 0.5, is it too small?

How i can cure it ?

Thanks a lot

Julien

**Comment by WPprodigy** [comment]
Does the store maybe have it set to use comma's as the decimal place? May need to do `0,50` if so.

**Comment by agenceldp** [comment]
Yes. About the calculation of the products works well.
You can find the parameters below : 

Options de devise : euro 
Position de la devise : droite
Séparateur milliers : .
Séparateur décimal : ,
Nombre de décimales : 2

**Comment by WPprodigy** [comment]
Gotcha. So if you enter a fee of `0,50`, does it work?

**Comment by agenceldp** [comment]
Exactly. the settings of the decimals is good but despite everything when i enter a fee : 0.50€ it does not calculate it.

**Comment by WPprodigy** [comment]
Sorry to keep asking, but I'm still a bit unclear so to clarify - you are using a COMMA when typing in the fee?

A screenshot of the settings area would help as well.

**Comment by agenceldp** [comment]
When I enter 0,5€ (with a comma) it doesn't work, whereas if I enter 1€, the calculation is done.

![Capture d’écran 2021-07-06 à 10 28 56](https://user-images.githubusercontent.com/86738851/124568492-4532d100-de45-11eb-89bb-274b61c1ccb4.png)

**Comment by WPprodigy** [comment]
Hmm, I see. I'm pretty sure something is conflicting with it, perhaps some rounding logic. If you have a test site/staging - trying disabling other plugins except for just WooCommerce & Product Fees. And switch to a default WordPress theme as well.

![fee](https://user-images.githubusercontent.com/8536129/124700338-2fa7c080-deb2-11eb-8fd5-53a3841c90fa.png)
![settings](https://user-images.githubusercontent.com/8536129/124700339-2fa7c080-deb2-11eb-9764-dc6b129fc784.png)
![cart](https://user-images.githubusercontent.com/8536129/124700340-30405700-deb2-11eb-950a-59e0c148cbd7.png)

**Comment by agenceldp** [comment]
Really i don't understand, i change the theme and disable all the extensions and despite i don't see the 0,5 fee...

**Comment by WPprodigy** [comment]
I'm really not sure either then :/. Perhaps take a look at rounding settings / number of decimals, as well as the tax classes you may be applying too fees.

And then this is the simplest distilled version of adding a fee:

```
add_action( 'woocommerce_cart_calculate_fees', function( $cart ) {
	$cart->add_fee( 'Example Fee', '0.50', false, '' );
}, 15 );
```

Could put that in your theme's function.php file and see what happens.
